client/src/main/java/org/eclipse/hono/client/HonoClient.java

```diff
@@ -117,13 +117,48 @@ void getOrCreateEventSender(
      * @param tenantId The tenant to consume data for.
      * @param telemetryConsumer The handler to invoke with every message received.
      * @param creationHandler The handler to invoke with the outcome of the operation.
+     * @param detachHandler The handler invoked on detached (not closed) link.
      * @throws NullPointerException if any of the parameters is {@code null}.
      */
+    void createTelemetryConsumer(
+            String tenantId,
+            Consumer<Message> telemetryConsumer,
+            Handler<AsyncResult<MessageConsumer>> creationHandler,
+            Handler<AsyncResult<Void>> detachHandler);
+
+    /**
+     * Creates a new consumer of telemetry data for a tenant.
+     *
+     * @param tenantId The tenant to consume data for.
+     * @param telemetryConsumer The handler to invoke with every message received.
+     * @param creationHandler The handler to invoke with the outcome of the operation.
+     * @throws NullPointerException if any of the parameters is {@code null}.
+     */
+    @Deprecated
```

Comment:
I think we do not need to deprecate this variant. Why not leave it up to clients to choose which variant they want to use? Being forced to pass in `null` to a method invocation never looks good in code FMPOV ...

Reply:
I think the question is if there is a good use case to use the API without this detach handler. My idea was to eliminate the deprecated methods, because without this handlers and a proper reaction there could always be a problem, which could not be healed without this handler. I think we should not provide an API, if we know, the usage will end in problems.